packages/compartment-mapper/src/node-powers.js

```diff
@@ -137,6 +165,53 @@ const makeReadPowersSloppy = ({ fs, url = undefined, crypto = undefined }) => {
     canonical,
     computeSha512,
     requireResolve,
+    isAbsolute,
+  };
+};
+
+/**
+ * Creates {@link ReadPowers} for dynamic module support
+ *
+ * @param {object} args
+ * @param {FsAPI} args.fs
+ * @param {UrlAPI} [args.url]
+ * @param {CryptoAPI} [args.crypto]
+ * @param {PathAPI} [args.path]
+ * @returns {MaybeReadPowers & SyncReadPowers}
+ */
+export const makeSyncReadPowers = ({
+  fs,
+  url = undefined,
+  crypto = undefined,
+  path = undefined,
+}) => {
+  const powers = makeReadPowersSloppy({ fs, url, crypto, path });
+  const fileURLToPath = powers.fileURLToPath || fakeFileURLToPath;
+  const isAbsolute = powers.isAbsolute || fakeIsAbsolute;
+
+  /**
+   * @type {MaybeReadSyncFn}
+   */
+  const maybeReadSync = location => {
+    const filepath = fileURLToPath(location);
+    try {
+      return fs.readFileSync(filepath);
+    } catch (error) {
+      if (
+        'code' in error &&
+        (error.code === 'ENOENT' || error.code === 'EISDIR')
```

Comment:
premature optimization thought: `EISDIR` is a hint to look at location/index.js and skip other candidates, but we have no good way to pass that information up to the loop that's iterating candidates.
I'm pretty sure it wouldn't be worth the complexity.

_will delete later_

Reply:
I was just adapting the logic in `maybeRead`.  I don't know the reason for checking `EISDIR`; I wouldn't otherwise.

¯\\_(ツ)_/¯

Reply:
the logic is - if this fails for any other reason, it's a failure that we don't want to silence. These are the two errors that are expected in the process.

Reply:
It’s a good point, @naugtur and I also don’t have any ideas on how to thread the hint, especially since some platforms have no way to communicate that hint. This way we at least have the same path regardless of backing storage.